frontend/packages/knative-plugin/src/components/add/EventSourceForm.tsx

```diff
@@ -31,28 +27,30 @@ const EventSourceForm: React.FC<FormikProps<FormikValues> & OwnProps> = ({
   namespace,
   projects,
 }) => (
-  <Form className="co-deploy-image" onSubmit={handleSubmit}>
+  <FlexForm onSubmit={handleSubmit}>
     <EventSourcesSelector eventSourceList={useEventSourceList(namespace)} />
-    {values.type === EventSources.CronJobSource && <CronJobSection />}
-    {values.type === EventSources.SinkBinding && <SinkBindingSection />}
-    {values.type === EventSources.ApiServerSource && <ApiServerSection />}
-    {values.type === EventSources.KafkaSource && <KafkaSourceSection />}
-    <SinkSection namespace={namespace} />
-    <AppSection
-      project={values.project}
-      noProjectsAvailable={projects?.loaded && _.isEmpty(projects.data)}
-    />
+    <EventSourceSection eventSourceType={values.type} />
+    {isKnownEventSource(values.type) && (
+      <>
+        <SinkSection namespace={namespace} />
+        <AppSection
+          project={values.project}
+          noProjectsAvailable={projects?.loaded && _.isEmpty(projects.data)}
+          flexLayout
+        />
+      </>
+    )}
     {values.type === EventSources.KafkaSource && <AdvancedSection />}
```

Comment:
@nemesis09 can we move all of these in `EventSourceSection` as you already have it now

Reply:
done